### PR TITLE
Travis simplifications

### PR DESCRIPTION
**Pull request by hannesa2**
It's about modernize this repo.

## In general 
Because of this multi git submodule setup, it's needs several PR on multiple repos, but as it looks like, the maintainer(s) keep not that much attention on it

## What I did:
* update to most recent build tools
* update to most recent support lib
* update to most recent other libs, where it was possible
* make Travis work with it
* switch temporary to updated forked git submodules
  (as soon, as the PR's there are merged, this can be removed)
* removed a workaround commit for supportLib 23 
  (btw, the end target should be AndroidX)
* (temporary ?) removed `witness` Gradle plugin
  - It can't handle most recent build tools
  - checks are not [perfect](https://github.com/signalapp/gradle-witness/issues/33)
  - last commit is from 2014, and last closed issue from 2015: It's the definition of **dead**
* But now tests make [issues](https://travis-ci.org/hannesa2/open-keychain/builds/560172020)
image

## I'm out of ideas.
I'm out of ideas, maybe update of Roboelectric makes sense ? But this needs AndroidX, as far as I observed.
*  I made a [branch](https://github.com/hannesa2/open-keychain/tree/AndroidX) with AndroidX, but this generates other additional issues, like update of https://github.com/square/sqldelight/releases, which causes new issues....

Maybe you can help me, to solve it, or you can uses this PR as basis to continue this work. Or close this PR unmerged

**Without help, I'm done here at this point**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "extern/openpgp-api-lib"]
 	path = extern/openpgp-api-lib
-	url = https://github.com/open-keychain/openpgp-api.git
+	url = https://github.com/hannesa2/openpgp-api.git
```
Comment (hannesa2) [inline .gitmodules:3]: Please merge the PR there and I can switch back to https://github.com/open-keychain/openpgp-api.git

**Comment by Valodim** [comment]
Hi @hannesa2. I did some major cleanup work in #2553 after some time away from the project. I think I covered everything from this PR, so I'll close this one. If you find something more feel free to open another PR.

**Comment by hannesa2** [comment]
I'm fine with all. For me the most important thing is keep this repo alive

**Comment by hannesa2** [comment]
I rebased it to #2553 
So it ends up in just Travis simplifications

**Comment by Valodim** [comment]
:+1:

**Comment by hannesa2** [comment]
This Travis is that much broken, I vote to remove it complete and go with https://github.com/open-keychain/open-keychain/pull/2556 only

**Comment by hannesa2** [comment]
In parallel I build on my fork to see results
https://github.com/hannesa2/open-keychain/pull/3

because here no CI is running at all, and you don't see how broken all is

**Pull request merged**